toptal/picasso
# [FX-5716] Use icon for dropdown arrows

[FX-5716](https://toptal-core.atlassian.net/browse/FX-5716)

### Description

This pull request aligns dropdown arrows with [BASE design](https://www.figma.com/design/5SCTOPrCDcHuk5We091GBn/Product-Library?node-id=297-13063&t=gy7cWXCe1lOCfuib-4).

Confirmed with @whymancreates [here](https://toptal-core.slack.com/archives/CCC3GP6CC/p1723113814084709).

### Details

**How it used to work before**

The triangle arrow was in fact a rectangle with top border being white and other borders being transparent to form the triangle.

<img width="1104" alt="Screenshot 2024-08-08 at 18 46 40" src="https://github.com/user-attachments/assets/598d9086-e5aa-429f-9d63-1858ed00ab21">

**Options**

There are three ways to solve the problem

1. modify existing "rectangle borders that form the triangle" approach to make it look like the icon in BASE
2. use existing icon `ArrowDownMinor16`
3. use custom icon (in fact, `ArrowDownMinor16`, but with modified `viewBox`)

---

Approach (1) is too hacky, so it was abandoned right from the start.

Approach (2) looks like the most straightforward one, but the problem is that Picasso icons have whitespace around the actual paths, so it heavily affects the calculations of space between elements in the dropdown.

| Icon component | Icon paths |
|-----------------|--------------|
| <img width="244" alt="Screenshot 2024-08-08 at 18 53 00" src="https://github.com/user-attachments/assets/7d9fae23-7ed9-4ca7-bead-2ff0864b8b7c"> | <img width="170" alt="Screenshot 2024-08-08 at 18 53 04" src="https://github.com/user-attachments/assets/2afcaf1b-a87e-4007-9a8c-b933ae762451"> |

In order to deal with extra whitespace in icon, negative margins need to be used to cut away this whitespace. However, in some cases negative margins still affect the overall layout, so all elements are shifted like in the visual test below

<img width="886" alt="Screenshot 2024-08-08 at 18 58 16" src="https://github.com/user-attachments/assets/13524b32-f8fe-469a-8174-930bd828653e">

However, there is another way to deal with SVG whitespace – modification of `viewBox`. After adjusting it to ignore the whitespace, there is no need in negative margins, so the overall layout is more clear. Additionally, there are no layout shifts affected by margins as you can see in visual test changes – differences are only about the arrow itself, no surroundings were affected.

<img width="900" alt="Screenshot 2024-08-08 at 20 21 59" src="https://github.com/user-attachments/assets/5095b782-ae18-43a3-a985-9446dc05535a">


Considering the fact that the Icons API should not allow modification of `viewBox` (there might be problems with overall calculations), it was decided to have a local copy of `ArrowDownMinor16` SVG with custom modifications needed particularly for this case.


### How to test


- Test [temploy](https://picasso.toptal.net/fx-5716-dropdown-arrows-do-not-match-with-base) in various browsers (feel free to use corporate account at https://browserstack.com)
- Check the visual changes, they should be only about arrow itself
- Compare Staff Portal [temploy](https://t-staff-portal-pr-13501.toptal.net/talents) (in https://github.com/toptal/staff-portal/pull/13501) and `staging` https://staff-portal.toptal.net/talents

### Screenshots

Before and after

### Development checks

- [N/A] Make sure that additions and changes on the design follow [Toptal's BASE design](https://design.toptal.net/), and it's been already discussed with designers at #-base-core
- [N/A] Annotate all `props` in component with documentation
- [N/A] Create `examples` for component
- [N/A] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).

> All **_development checks_** should be done and set checked to pass the
> **GitHub Bot: TODOLess** action

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run package:alpha-release` - Release alpha version
- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>


[FX-5716]: https://toptal-core.atlassian.net/browse/FX-5716?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
